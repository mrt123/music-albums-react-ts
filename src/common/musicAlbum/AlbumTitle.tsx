const getStyle = () => {
  return {
    color: "#d8ed5c9c",
    fontWeight: "bold",
    fontSize: "18px",
    fontFamily: "'Courier New', monospace",
    width: "330px",
    textOverflow: "ellipsis",
    overflow: "hidden",
    whiteSpace: "nowrap" as "nowrap",
  };
};

interface Props {
  children: React.ReactNode;
}

const AlbumTitle: React.FC<Props> = ({ children }) => {
  return <div style={getStyle()}>{children}</div>;
};

export default AlbumTitle;
