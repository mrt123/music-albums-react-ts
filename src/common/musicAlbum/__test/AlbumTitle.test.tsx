import { render, RenderResult } from "@testing-library/react";
import exp from "constants";
import AlbumTitle from "../AlbumTitle";

describe("AlbumTitle.tsx", () => {
  it("matches snapshot for string children", () => {
    const renderResult: RenderResult = render(
      <AlbumTitle>My Greatest Hits</AlbumTitle>
    );

    const container = renderResult.container;

    expect(container).toMatchSnapshot();
  });

  it("matches snapshot for a complex children elements", () => {
    const { container } = render(
      <AlbumTitle>
        <div id="additional_div">text inside another div</div>
      </AlbumTitle>
    );
    expect(container).toMatchSnapshot();
  });
});
