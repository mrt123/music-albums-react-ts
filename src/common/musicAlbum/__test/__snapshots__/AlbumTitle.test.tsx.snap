// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumTitle.tsx matches snapshot for a complex children elements 1`] = `
<div>
  <div
    style="color: rgba(216, 237, 92, 0.612); font-weight: bold; font-size: 18px; font-family: 'Courier New', monospace; width: 330px; text-overflow: ellipsis; overflow: hidden; white-space: nowrap;"
  >
    <div
      id="additional_div"
    >
      text inside another div
    </div>
  </div>
</div>
`;

exports[`AlbumTitle.tsx matches snapshot for string children 1`] = `
<div>
  <div
    style="color: rgba(216, 237, 92, 0.612); font-weight: bold; font-size: 18px; font-family: 'Courier New', monospace; width: 330px; text-overflow: ellipsis; overflow: hidden; white-space: nowrap;"
  >
    My Greatest Hits
  </div>
</div>
`;
